### PR TITLE
wasi-nn: simplify cmake and headers location

### PR DESCRIPTION
**Pull request by tonibofarull**
## Proposed Changes

* Public headers inside `wasi-nn/include`.
* Cmake files in `cmake`.
* `linux` iwasm contains `${TENSORFLOW_LIB}`. iwasm can enable wasi-nn.

### CODE REVIEW

**Comment by tonibofarull** [comment]
@ayakoakasaka please review!

File: core/iwasm/libraries/wasi-nn/cmake/iwasm_helper.cmake [hunk 1]
```diff
@@ -0,0 +1,43 @@
+if (NOT DEFINED WAMR_BUILD_PLATFORM)
```
Comment (wenyongh) [inline core/iwasm/libraries/wasi-nn/cmake/iwasm_helper.cmake:1]: Should add license header

**Review by wenyongh — COMMENTED** [review]
LGTM

File: product-mini/platforms/linux/CMakeLists.txt [hunk 2]
```diff
@@ -155,7 +155,7 @@ set_target_properties (iwasm PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 install (TARGETS iwasm DESTINATION bin)
 
-target_link_libraries (iwasm vmlib ${LLVM_AVAILABLE_LIBS} ${UV_A_LIBS} -lm -ldl -lpthread)
+target_link_libraries (iwasm vmlib ${LLVM_AVAILABLE_LIBS} ${UV_A_LIBS} ${TENSORFLOW_LIB} -lm -ldl -lpthread)
```
Comment (ayakoakasaka) [inline product-mini/platforms/linux/CMakeLists.txt:158]: Considering that wasi-nn will support other than Tensorflow in the future, isn't NN_LIBS better as a MACRO name?

File: product-mini/platforms/linux/CMakeLists.txt [hunk 3]
```diff
@@ -155,7 +155,7 @@ set_target_properties (iwasm PROPERTIES POSITION_INDEPENDENT_CODE ON)
...
 
 add_library (libiwasm SHARED ${WAMR_RUNTIME_LIB_SOURCE})
```
Comment (ayakoakasaka) [inline product-mini/platforms/linux/CMakeLists.txt:160]: Don't we need to add "$NN_LIBS"(a.k.a ${TENSORFLOW_LIB})  for libiwasm as well?

Comment (tonibofarull) [reply product-mini/platforms/linux/CMakeLists.txt:160]: you are right! https://github.com/bytecodealliance/wasm-micro-runtime/pull/2308/commits/25f593a342d9e004635f6ccd303bfbcaaa90509b

File: product-mini/platforms/linux/CMakeLists.txt (hunk 2 above)
Comment (tonibofarull) [reply product-mini/platforms/linux/CMakeLists.txt]: good catch!

Comment (tonibofarull) [reply product-mini/platforms/linux/CMakeLists.txt]: https://github.com/bytecodealliance/wasm-micro-runtime/pull/2308/commits/25f593a342d9e004635f6ccd303bfbcaaa90509b

**Pull request merged**